docs/adr/0009-ed25519-semantics.md

```diff
@@ -0,0 +1,124 @@
+# ADR 0009: Ed25519 Signature Verification Semantics
+
+## Changelog
+
+- 2021-05-10: Initial version
+
+## Status
+
+Informative
+
+## Context
+
+> In programming, it's often the buts in the specification that kill you.
+>
+> -- Boris Beizer
+
+For a large host of reasons, mostly historical, there are numerous definitions
+of "Ed25519 signature validation" in the wild, which have the potential to
+be mutually incompatible.  This ADR serves to provide a rough high-level
+overview of the issue, and to document the current definition of "Ed25519
+signature verification" as used by Oasis Core.
+
+## Decision
+
+The Oasis Core consensus layer (and all of the Go components) currently uses
+the following Ed25519 verification semantics.
+
+- Non-canonical s is rejected (`s < L`)
```

Comment:
for those unfamiliar, it's not clear that `s < L` is the test that causes rejection or is a requirement.  maybe reword the parathetical as "(i.e., require `s < L`)" (or something equiv)?